### ISSUE TITLE
Feature Request: Epidemiology-inspired rolling window quantiles

### ISSUE THREAD

**Issue by aweyant** [aweyant/weatherAndClimateUtils#2]
Rolling window quantiles of width w can be calculated on all observations in years leading up to a date of interest.

e.g. we may be interested in 90th ptile of temperature on 23 July 2024. This can be calculated as the empirical 90th percentile of temperatures on {20,21,...,26} July in all years before 2024.

@axdey this correct?

**Comment by arnabkdey** [comment]
Yes - plus the percentile will be specific to the geographic region of interest e.g. Zip code
